### ISSUE TITLE
Inference error on handling initial times

### ISSUE THREAD

**Issue by oliverwm1** [ai2cm/ace#479]
An [inference run](https://beaker.org/ex/01JBHZ0EWTRP8M0P36EB0SPEZ9/tasks/01JBHZ0EX02461HSS7BQ4JD3SY/job/01JBHZ0F1ZASNVR237AW3S0MAR) failed with. This new code, introduced in https://github.com/ai2cm/full-model/pull/1288, adds assumptions about dimension names which cause production inference runs to fail. FYI @mcgibbon 

```
2024-10-31T19:08:36.018683558Z 2024-10-31 19:08:36,018 - root - INFO - Inference: processing window spanning 0 to 5 steps.
2024-10-31T19:08:36.234344048Z Traceback (most recent call last):
2024-10-31T19:08:36.235107033Z   File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
2024-10-31T19:08:36.235156410Z     return _run_code(code, main_globals, None,
2024-10-31T19:08:36.235498451Z   File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
2024-10-31T19:08:36.235534712Z     exec(code, run_globals)
2024-10-31T19:08:36.235791475Z   File "/full-model/fme/fme/ace/inference/__main__.py", line 16, in <module>
2024-10-31T19:08:36.235802637Z     main(yaml_config=args.yaml_config, segments=args.segments)
2024-10-31T19:08:36.236098287Z   File "/full-model/fme/fme/ace/inference/inference.py", line 211, in main
2024-10-31T19:08:36.236109489Z     return run_inference_from_config(config)
2024-10-31T19:08:36.236439115Z   File "/full-model/fme/fme/ace/inference/inference.py", line 267, in run_inference_from_config
2024-10-31T19:08:36.236450587Z     run_inference(
2024-10-31T19:08:36.236754924Z   File "/full-model/fme/fme/ace/inference/loop.py", line 234, in run_inference
2024-10-31T19:08:36.236766386Z     data=ic_filled, times=initial_times.expand_dims("time", axis=1)
2024-10-31T19:08:36.237114990Z   File "/usr/local/lib/python3.10/dist-packages/xarray/core/dataarray.py", line 2681, in expand_dims
2024-10-31T19:08:36.237126963Z     ds = self._to_temp_dataset().expand_dims(
2024-10-31T19:08:36.237429737Z   File "/usr/local/lib/python3.10/dist-packages/xarray/core/dataset.py", line 4734, in expand_dims
2024-10-31T19:08:36.237441840Z     raise ValueError(f"Dimension {d} already exists.")
2024-10-31T19:08:36.237778340Z ValueError: Dimension time already exists.
```

**Comment by oliverwm1** [comment]
I think the bigger issue here is that it was a mistake to use `xr.DataArray` as the type for the `.times` attribute of `BatchData`. It should simply by a numpy array, and then we don't have to worry about annoying xarray dimension names. When needing xarray functionality (like to compute seasonal means, etc.) we can always make an xarray DataArray out of the numpy array. But I think it would be better to stay more low level in BatchData, similar to the `.data` attr.

**Comment by mcgibbon** [comment]
Agreed, I'll look at fixing this by making that change.

I guess this happens because in our inference data the sample dim is called "time"?

**Comment by oliverwm1** [comment]
> I guess this happens because in our inference data the sample dim is called "time"?

Exactly.

**Comment by mcgibbon** [comment]
For the sake of fixing main fast I'll make a smaller fix PR first.

**Comment by mcgibbon** [comment]
Should be resolved by ai2cm/full-model#1309
